### PR TITLE
Watchable support

### PR DESCRIPTION
**Pull request by novakov-alexey-zz**
- Add support of Watchable for all implemented K8s resources (see more: https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes)
- Run all test cases within the scalatest suite in parallel by using `ParallelTestExecution` trait. This gives huge speedup. (30 seconds vs 5 minutes on 3 Nodes kubernetes cluster). All `<Resources>ApiTest` suites are sharing one namespace such as "${resourceName.toLowerCase}", except the NamespaceApiTest, since the point of this class is to test namespace operation and not the resources inside it.

Should be checked after PR #8 is merged to master

### CODE REVIEW

File: project/SwaggerModelGenerator.scala [hunk 1]
```diff
@@ -42,7 +45,7 @@ object SwaggerModelGenerator extends AutoPlugin {
       "io.k8s.kubernetes.pkg.apis.batch.v1",
       "io.k8s.kubernetes.pkg.apis.networking.v1"
     )
-    allowedPrefixes.exists(className.startsWith)
+    allowedPrefixes.exists(className.startsWith) && !skipClases.contains(className)
```
Comment (joan38) [inline project/SwaggerModelGenerator.scala]: Typo in `skipClases`, should be `skipClasses`?

File: src/test/scala/com/goyeau/kubernetes/client/api/ContextProvider.scala [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.goyeau.kubernetes.client.api
+
+import cats.effect.{Blocker, ContextShift, IO, Timer}
+
+import scala.concurrent.ExecutionContext
+
+trait ContextProvider {
+
+  implicit lazy val ioTimer: Timer[IO]   = IO.timer(ExecutionContext.global)
+  implicit lazy val cs: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
```
Comment (joan38) [inline src/test/scala/com/goyeau/kubernetes/client/api/ContextProvider.scala]: Let's name them `timer` and `contextShift`

File: project/SwaggerModelGenerator.scala (hunk 1 above)
Comment (novakov-alexey-zz) [reply project/SwaggerModelGenerator.scala]: Fixed

File: src/test/scala/com/goyeau/kubernetes/client/api/ContextProvider.scala (hunk 2 above)
Comment (novakov-alexey-zz) [reply src/test/scala/com/goyeau/kubernetes/client/api/ContextProvider.scala]: Done. I have also removed redundant `ContextShift` and `Timer` from *Test clases.

**Comment by joan38** [comment]
I merged #11 so you can rebase on master

**Comment by novakov-alexey-zz** [comment]
New version of this PR is created here #12

**Pull request closed without merging**